gomnd regression in 1.35.2

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

I'm trying to upgrade my project from 1.33 to 1.35 and getting weird random behavior of `gomnd` where magic constants are treated as not magic triggering `nolint` violation. 

See the pull request to upgrade: https://github.com/kopia/kopia/pull/786

It is  failing GH checks: https://github.com/kopia/kopia/pull/786/checks?check_run_id=1715239415
Notice the linter passes on macOS, Windows and using provided GH Action but fails on Linux:

I can reproduce this with https://github.com/jkowalski/kopia/commit/f0f84416e8555059ad80c07a69c3009c14860d76:

```
$ GOOS=linux GOARCH=amd64 make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 
repo/encryption/deprecated_ctr_encryptor.go:83:110: directive `//nolint:gomnd` is unused for linter gomnd (nolintlint)
	Register("AES-192-CTR", "DEPRECATED: AES-192 in CTR mode", true, newCTREncryptorFactory(24, aes.NewCipher)) //nolint:gomnd
	                                                                                                            ^
make: *** [lint] Error 1

$ GOOS=darwin GOARCH=amd64 make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 

$ GOOS=windows GOARCH=amd64 make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 
```

If you look at the history of this PR you'll find couple more spurious failures with `gomnd` that I have also fixed, such as https://github.com/kopia/kopia/pull/786/commits/1c56fb7bd11912f043fe7c0225cc0655660b00cf

Also, did some tests on different versions all using the same commit:

https://github.com/kopia/kopia/commit/1f3b8d4da4bdf76cd6060ed652bafeb372af5bfe

Using 1.35.2:

```
$  GOARCH=amd64 GOOS=linux make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 

$ GOARCH=amd64 GOOS=darwin make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 

$ GOARCH=amd64 GOOS=windows make lint
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.35.2/golangci-lint --deadline 300s run 
repo/splitter/splitter.go:45:35: directive `//nolint:gomnd` is unused for linter gomnd (nolintlint)
	"FIXED-8M": Fixed(megabytes(8)), //nolint:gomnd
```

Using 1.34.1 (only changed `tools/tools.mk`) linux/windows have the same failures (all seem to be bugs in `thelper` which does not recognize `ctx`) but `darwin` has one spurious failure:

```
$ GOARCH=amd64 GOOS=linux make lint 
# same as $ GOARCH=amd64 GOOS=windows make lint                                                                                                                                              
/Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.34.1/golangci-lint --deadline 300s run 
internal/repotesting/repotesting.go:94:1: parameter *testing.T should be the first (thelper)
func (e *Environment) Close(ctx context.Context, t *testing.T) {
^
repo/repository_test.go:200:1: parameter *testing.T should be the first (thelper)
func writeObject(ctx context.Context, t *testing.T, rep repo.Repository, data []byte, testCaseID string) object.ID {
^
repo/repository_test.go:216:1: parameter *testing.T should be the first (thelper)
func verify(ctx context.Context, t *testing.T, rep repo.Repository, objectID object.ID, expectedData []byte, testCaseID string) {
^
make: *** [lint] Error 1

$ GOARCH=amd64 GOOS=darwin make lint                                                                                                                                               /Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.34.1/golangci-lint --deadline 300s run 
internal/repotesting/repotesting.go:94:1: parameter *testing.T should be the first (thelper)
func (e *Environment) Close(ctx context.Context, t *testing.T) {
^
repo/blob/sftp/sftp_storage_test.go:43:1: parameter *testing.T should be the first (thelper)
func deleteBlobs(ctx context.Context, t *testing.T, st blob.Storage) {
^
repo/blob/sftp/sftp_storage_test.go:53:1: parameter *testing.T should be the first (thelper)
func createSFTPStorage(ctx context.Context, t *testing.T, embed bool) (blob.Storage, error) {
^
repo/splitter/splitter.go:45:35: directive `//nolint:gomnd` is unused for linter gomnd (nolintlint)
	"FIXED-8M": Fixed(megabytes(8)), //nolint:gomnd
```

On 1.33.0 all three {linux,darwin,windows}/amd64 return the same errors (seems like a bug since 0o555 is clearly a magic constant)

```
$ /Users/jarek/Projects/Kopia/tools/.tools/golangci-lint-1.33.0/golangci-lint --deadline 300s run 
WARN [runner/nolint] Found unknown linters in //nolint directives: forbidigo 
snapshot/snapshotfs/all_sources.go:33:28: directive `// nolint:gomnd` is unused for linter gomnd (nolintlint)
	return 0o555 | os.ModeDir // nolint:gomnd
	                          ^
snapshot/snapshotfs/source_directories.go:29:28: directive `// nolint:gomnd` is unused for linter gomnd (nolintlint)
	return 0o555 | os.ModeDir // nolint:gomnd
	                          ^
snapshot/snapshotfs/source_snapshots.go:31:28: directive `// nolint:gomnd` is unused for linter gomnd (nolintlint)
	return 0o555 | os.ModeDir // nolint:gomnd
```

I created a PR to fix a concurrent access https://github.com/tommy-muehle/go-mnd/pull/19

```console
$ mnd  ./...
./repo/content/content_index_recovery.go:64:9: Magic number: 255, in <condition> detected
./repo/content/content_index_recovery.go:84:23: Magic number: 5, in <condition> detected
./repo/content/blob_crypto.go:25:57: Magic number: 2, in <operation> detected
./repo/content/builder.go:175:54: Magic number: 16, in <operation> detected
./repo/content/builder.go:190:51: Magic number: 8, in <operation> detected
./repo/content/content_manager_lock_free.go:130:78: Magic number: 2, in <operation> detected
./repo/content/content_read_manager.go:77:19: Magic number: 100, in <operation> detected
./repo/splitter/splitter.go:43:30: Magic number: 2, in <argument> detected
./repo/splitter/splitter.go:44:30: Magic number: 4, in <argument> detected
./repo/splitter/splitter.go:45:30: Magic number: 8, in <argument> detected
./repo/splitter/splitter.go:48:62: Magic number: 2, in <argument> detected
./repo/splitter/splitter.go:49:62: Magic number: 4, in <argument> detected
./repo/splitter/splitter.go:50:62: Magic number: 8, in <argument> detected
./repo/splitter/splitter.go:53:66: Magic number: 2, in <argument> detected
./repo/splitter/splitter.go:54:66: Magic number: 4, in <argument> detected
./repo/splitter/splitter.go:55:66: Magic number: 8, in <argument> detected
./repo/splitter/splitter.go:58:17: Magic number: 4, in <argument> detected
./repo/splitter/splitter.go:58:22: Magic number: 20, in <argument> detected
./repo/splitter/splitter.go:62:51: Magic number: 4, in <argument> detected
./repo/splitter/splitter_buzhash32.go:106:23: Magic number: 2, in <operation> detected
./repo/splitter/splitter_buzhash32.go:107:23: Magic number: 2, in <operation> detected
./repo/splitter/splitter_rabinkarp64.go:104:30: Magic number: 2, in <operation> detected
./repo/splitter/splitter_rabinkarp64.go:104:41: Magic number: 2, in <operation> detected
./repo/splitter/splitter.go:66:15: Magic number: 20, in <return> detected
./repo/object/object_reader.go:127:30: Magic number: 2, in <operation> detected
./fs/cachefs/cache.go:176:24: Magic number: 1000, in <assign> detected
./fs/cachefs/cache.go:177:24: Magic number: 100000, in <assign> detected
./repo/format_block.go:279:49: Magic number: 8, in <argument> detected
./repo/format_block.go:281:43: Magic number: 8, in <argument> detected
./repo/initialize.go:98:62: Magic number: 20, in <argument> detected
./repo/initialize.go:98:66: Magic number: 20, in <argument> detected
./repo/open.go:161:20: Magic number: 20, in <operation> detected
./repo/open.go:161:26: Magic number: 20, in <operation> detected
./snapshot/policy/retention_policy.go:170:31: Magic number: 7, in <argument> detected
./snapshot/policy/retention_policy.go:178:22: Magic number: 10, in <argument> detected
./snapshot/policy/retention_policy.go:179:22: Magic number: 48, in <argument> detected
./snapshot/policy/retention_policy.go:180:22: Magic number: 7, in <argument> detected
./snapshot/policy/retention_policy.go:181:22: Magic number: 4, in <argument> detected
./snapshot/policy/retention_policy.go:182:22: Magic number: 24, in <argument> detected
./snapshot/policy/retention_policy.go:183:22: Magic number: 3, in <argument> detected
./snapshot/snapshotfs/upload.go:712:16: Magic number: 100, in <argument> detected
./snapshot/snapshotfs/all_sources.go:33:9: Magic number: 0o555, in <return> detected
./snapshot/snapshotfs/source_directories.go:29:9: Magic number: 0o555, in <return> detected
./snapshot/snapshotfs/source_snapshots.go:31:9: Magic number: 0o555, in <return> detected
./snapshot/snapshotfs/repofs.go:220:16: Magic number: 0o555, in <assign> detected
./snapshot/snapshotfs/repofs.go:266:16: Magic number: 0o644, in <assign> detected
./snapshot/snapshotfs/source_snapshots.go:83:17: Magic number: 0o555, in <assign> detected
./repo/maintenance/content_rewrite.go:138:95: Magic number: 100, in <argument> detected
./repo/maintenance/maintenance_params.go:46:19: Magic number: 24, in <assign> detected
./internal/server/htmlui_fallback.go:161:24: Magic number: 2147484068, in <argument> detected
./internal/server/htmlui_fallback.go:186:75: Magic number: 420, in <argument> detected
./internal/server/htmlui_fallback.go:186:100: Magic number: 1574253768, in <argument> detected
./internal/server/htmlui_fallback.go:300:68: Magic number: 0o755, in <argument> detected
./internal/server/htmlui_fallback.go:186:52: Magic number: 358, in <assign> detected
./repo/blob/b2/b2_storage.go:115:47: Magic number: 1e6, in <argument> detected
./cli/command_benchmark_splitters.go:96:29: Magic number: 1e6, in <argument> detected
./cli/command_benchmark_splitters.go:112:29: Magic number: 1e6, in <argument> detected
./cli/command_snapshot_verify.go:174:15: Magic number: 100, in <argument> detected
./cli/command_repository_sync.go:214:37: Magic number: 8, in <operation> detected
./cli/command_snapshot_create.go:118:63: Magic number: 20, in <operation> detected
./cli/command_snapshot_estimate.go:123:45: Magic number: 8, in <operation> detected
./cli/command_snapshot_estimate.go:123:49: Magic number: 1000000, in <operation> detected
./cli/command_repository_connect.go:51:56: Magic number: 20, in <assign> detected
./cli/command_repository_connect.go:52:64: Magic number: 20, in <assign> detected
./cli/command_snapshot_estimate.go:58:20: Magic number: 1e15, in <assign> detected
./cli/command_snapshot_estimate.go:59:20: Magic number: 1e12, in <assign> detected
./cli/command_snapshot_estimate.go:60:20: Magic number: 1e9, in <assign> detected
./cli/command_snapshot_estimate.go:61:20: Magic number: 1e6, in <assign> detected
./cli/command_snapshot_estimate.go:62:20: Magic number: 1e3, in <assign> detected
./examples/upload_download/setup_repository.go:51:24: Magic number: 100000000, in <assign> detected
./internal/blobtesting/asserts.go:36:32: Magic number: 2, in <argument> detected
./internal/blobtesting/concurrent.go:61:18: Magic number: 100, in <argument> detected
./internal/blobtesting/concurrent.go:129:18: Magic number: 100, in <argument> detected
./internal/mockfs/mockfs.go:290:10: Magic number: 0o777, in <assign> detected
./tests/perf_benchmark/process_results.go:111:20: Magic number: 2, in <condition> detected
./tests/tools/kopiarunner/kopia_snapshotter.go:247:19: Magic number: 2, in <condition> detected
./tests/testenv/cli_test_env.go:468:42: Magic number: 100, in <argument> detected
./tests/testenv/cli_test_env.go:514:26: Magic number: 2, in <argument> detected
./tests/testenv/cli_test_env.go:272:63: Magic number: 2, in <operation> detected
./tests/testenv/cli_test_env.go:274:64: Magic number: 2, in <operation> detected
./tests/testenv/faketimeserver.go:55:20: Magic number: 100, in <assign> detected
./tests/testingaction/main.go:92:20: Magic number: 2, in <condition> detected
```

As you can see the line `repo/splitter/splitter.go:45:35` is not is the report (on Linux) but I don't know why.

--- a/go.mod
+++ b/go.mod
@@ -63,7 +63,7 @@ require (
 	github.com/tetafro/godot v1.3.2
 	github.com/timakin/bodyclose v0.0.0-20190930140734-f7f2e9bca95e
 	github.com/tomarrell/wrapcheck v0.0.0-20200807122107-df9e8bcb914d
-	github.com/tommy-muehle/go-mnd v1.3.1-0.20201008215730-16041ac3fe65
+	github.com/tommy-muehle/go-mnd/v2 v2.3.1
 	github.com/ultraware/funlen v0.0.3
 	github.com/ultraware/whitespace v0.0.4
 	github.com/uudashr/gocognit v1.0.1

--- a/pkg/golinters/gomnd.go
+++ b/pkg/golinters/gomnd.go
@@ -1,7 +1,7 @@
 package golinters
 
 import (
-	mnd "github.com/tommy-muehle/go-mnd"
+	mnd "github.com/tommy-muehle/go-mnd/v2"
 	"golang.org/x/tools/go/analysis"
 
 	"github.com/golangci/golangci-lint/pkg/config"

